src/net/tcp/listener.rs

```diff
@@ -75,8 +76,12 @@ impl TcpListener {
     /// [`local_addr`]: #method.local_addr
     pub async fn bind<A: ToSocketAddrs>(addrs: A) -> io::Result<TcpListener> {
         let mut last_err = None;
+        let addrs = addrs
+            .to_socket_addrs()
+            .await
+            .context(|| String::from("could not resolve addresses"))?;
```

Comment:
It'd be great to know _which_ addresses couldn't be resolved. I feel this is one of the most common cases folks are left wondering about.

Reply:
~~Yeah, that's what I wanted as well, but we sadly can't do that without adding more constraints on `A` which would be a breaking change~~

Wrong actually! Since is `async_std::net::ToSocketAddrs`, not the trait from std! If you want to add that info, we can actually add a method like `#[doc(hidden)] fn debug_print_addr<D: Debug>(&self) -> Option<T> { None }` without breaking API!

Or, actually even better: Move the `.context` call into `to_socket_addrs`!

That'd be something for a different PR, though :)

Reply:
Actually, scratch all that. It was easier than expected and I added a new commit :)